libmate-desktop/mate-desktop-item.c

```diff
@@ -1810,63 +1811,67 @@ ditem_execute (const MateDesktopItem *item,
 	arg_ptr = make_args (file_list);
 
 #ifdef HAVE_STARTUP_NOTIFICATION
-	if (screen)
-		gdkdisplay = gdk_screen_get_display (screen);
-	else
-		gdkdisplay = gdk_display_get_default ();
-
-	sn_display = sn_display_new (GDK_DISPLAY_XDISPLAY (gdkdisplay),
-				     sn_error_trap_push,
-				     sn_error_trap_pop);
+	GdkDisplay *display = gdk_screen_get_display (gdk_screen_get_default());
+	if (GDK_IS_X11_DISPLAY (display))
+	{
+		if (screen)
+			gdkdisplay = gdk_screen_get_display (screen);
+		else
+			gdkdisplay = gdk_display_get_default ();
 
-	/* Only initiate notification if desktop file supports it.
-	 * (we could avoid setting up the SnLauncherContext if we aren't going
-	 * to initiate, but why bother)
-	 */
+		sn_display = sn_display_new (GDK_DISPLAY_XDISPLAY (gdkdisplay),
+					     sn_error_trap_push,
+					     sn_error_trap_pop);
 
-	startup_class = mate_desktop_item_get_string (item,
-						       "StartupWMClass");
-	if (startup_class ||
-	    mate_desktop_item_get_boolean (item, "StartupNotify")) {
-		const char *name;
-		const char *icon;
+		/* Only initiate notification if desktop file supports it.
+		 * (we could avoid setting up the SnLauncherContext if we aren't going
+		 * to initiate, but why bother)
+		 */
 
-		sn_context = sn_launcher_context_new (sn_display,
-						      screen ? gdk_x11_screen_get_screen_number (screen) :
-						      DefaultScreen (GDK_DISPLAY_XDISPLAY (gdkdisplay)));
+		startup_class = mate_desktop_item_get_string (item,
+							      "StartupWMClass");
+		if (startup_class ||
+		    mate_desktop_item_get_boolean (item, "StartupNotify")) {
+			const char *name;
+			const char *icon;
 
-		name = mate_desktop_item_get_localestring (item,
-							    MATE_DESKTOP_ITEM_NAME);
+			sn_context = sn_launcher_context_new (sn_display,
+							      screen ? gdk_x11_screen_get_screen_number (screen) :
+							      DefaultScreen (GDK_DISPLAY_XDISPLAY (gdkdisplay)));
 
-		if (name == NULL)
 			name = mate_desktop_item_get_localestring (item,
-								    MATE_DESKTOP_ITEM_GENERIC_NAME);
+								   MATE_DESKTOP_ITEM_NAME);
 
-		if (name != NULL) {
-			char *description;
+			if (name == NULL)
+				name = mate_desktop_item_get_localestring (item,
+									    MATE_DESKTOP_ITEM_GENERIC_NAME);
```

Comment:
One more indent issue.
```suggestion
				name = mate_desktop_item_get_localestring (item,
									   MATE_DESKTOP_ITEM_GENERIC_NAME);
```

Reply:
Fixed that one too, thanks for finding it